File: java/core/src/core/crypt/CryptorNone.java
/**
 * Author: Timothy Prepscius
 * License: GPLv3 Affero + keep my name in the code!
 */

package core.crypt;

import core.exceptions.CryptoException;

public class CryptorNone extends Cryptor
{

	@Override
	public byte[] encrypt(byte[] bytes) throws CryptoException 
	{
		return bytes;
	}

	@Override
	public byte[] decrypt(byte[] bytes) throws CryptoException 
	{
		return bytes;
	}

}

File: cpp/core/mailiverse/core/crypt/CryptorNone.cpp
/**
 * Author: Timothy Prepscius
 * License: BSD + keep my name in the code!
 */

#include "CryptorNone.h"
#include <sstream>
#include "mailiverse/utilities/Log.h"

using namespace mailiverse::core::crypt;
using namespace mailiverse;

Block CryptorNone::encrypt (const Block::Atom *begin, const Block::Atom *end)
{
	return Block(begin, end);
}

Block CryptorNone::decrypt (const Block::Atom *begin, const Block::Atom *end)
{
	return Block(begin, end);
}